src/Router.js

```diff
@@ -88,6 +88,9 @@ class Router extends Component {
     }
 
     Actions.callback = props => {
+      if (props.type && sceneTypeMapAction[props.type]) {
+        props.type = sceneTypeMapAction[props.type];
+      }
       if (this.props.dispatch) this.props.dispatch(props);
```

Comment:
My idea was slightly different - don't modify type (to avoid breaking existing apps), but just pass modified type to 'dispatch' method only